wyvern/experimentation/client.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+from .providers.eppo_provider import EppoExperimentationClient
+
+
+class ExperimentationClient:
+    def __init__(self, provider_name: str):
+        """
+        Initializes the ExperimentationClient with a specified provider.
+
+        Args:
+        - provider_name (str): The name of the experimentation provider (e.g., "eppo").
+        """
+        if provider_name == "eppo":
+            self.provider = EppoExperimentationClient()
+        else:
+            raise ValueError(f"Unsupported provider: {provider_name}")
+
+    def get_experiment_result(
+        self, experiment_id: str, entity_id: str, **kwargs
+    ) -> str:
+        """
+        Get the result (variant) for a given experiment and entity using the chosen provider.
+
+        Args:
+        - experiment_id (str): The unique ID of the experiment.
+        - entity_id (str): The unique ID of the entity.
+        - kwargs (dict): Any additional arguments to pass to the provider for targeting.
+
+        Returns:
+        - str: The result (variant) assigned to the entity for the specified experiment.
+        """
+        # TODO (kerem): What happens if the provider is down / returns None?
```

Comment:
I think the get_result should always enforce a "default" parameter in case when experimentation provider is down. it shouldn't be a blocker for the pipeline

Reply:
I'm not sure whether we should enforce a default value or not tho. wdyt?

Reply:
I think we should let them define it. But another way to do it is to change how get_result works:
```
def is_experiment_on(self, expected_value:str ,experiment_id: str, entity_id: str, **kwargs) -> bool:
        client = eppo_client.get_instance()
        result = client.get_assignment(entity_id, experiment_id, kwargs)
        return result == expected_value
```

Reply:
what is `is_experiment_on`? is `is_experiment_on` going to be an interface of `ExperimentationClient`?

Reply:
i don't understand how is_experiment_on helps if the provider (eppo in this case) is down. I still prefer the "default_value" solution.

Reply:
going with adding a `default_value` argument to get_result method